fix(ui-kit):Fix upload component to support displaying multiple files and images.

jatinrise opened the pull request:
- Component name: `image-uploader` `file-uploader` `video-uploader`

This PR includes the following changes:

1. Separated upload logic into three different files: image, file, and video, as each requires different props. Additionally, multiple file uploads are now supported with a configurable upload limit.
2. Made some updates to the tests.


<img width="470" alt="Screenshot 2025-05-03 at 14 33 46" src="https://github.com/user-attachments/assets/5d21e204-7aa8-4fd7-8daf-88720d5ec736" />
## React Best Practices Checklist

- [x] I have avoided **unnecessary re-renders** by using composition (`children` as props), whenever possible. This applies, in particular, to composite components (like lists of components, and the like).
- [x] **I have designed my components as server-side whenever possible**—I have not used `useState`, `useEffect`, `useContext`, or any other React hook that works with "use client" without a good reason.
- [x] **I have not put props in state**—in general, props should remain as props unless the component needs an `initialValue`, and judiciously manage its own independent state detached from the parent component.
    - If I have stored props in state, I have explained where exactly and justified it with a solid reason here below:
    
- [x] **I have avoided using Context** in any critical component in terms of performance
    - If I have used Context, I have explained where exactly and justified it with a solid reason here below:


## General Checklist

- [x] I have run the lint, build, and test commands locally, as explained in the README, before creating this PR to ensure the code is clean and functional.
- [x] My changes align with the **core models** that have been discussed and agreed upon.
- [x] The component design has been compared to the **Figma design** to ensure consistency.
- [x] The **component is fully responsive** and behaves as expected across different screen sizes.
- [x] The component **supports keyboard web accessibility** (if applicable).
- [x] I have written **unit tests** or **integration tests** to cover the component's functionality above the coverage threshold.
- [x] I have updated the **documentation** for the component, including usage examples and guidelines.
- [x] I have **commented** my code, particularly in complex or non-obvious areas.
- [x] The **component is properly exported** for use in other parts of the application.

fix #186

alebg reviewed (changes requested):
@jatinrise @anilrise11 So I see that most of the basic code comes from an old PR we merged. So I think it is a good time to improve it. Please take my comments below as a revision of what was already done, but being more strict regarding our guidelines, in particular given that this will be used in a crucial part of the app.

So, in general, the typing of what was already merged is rather opaque and confusing, and adding new things on top of this starts to feel inconsistent. At this point, I'd say that what was done needs a re-implementation (and all the places where used a refactor, as needed), as opposed to piling new code on top. So here's more concrete comments:

- Why is the name of the upload props "Enhanced"? If not meaningful, please omit naming things in a confusing way

- I don't agree with the `type` prop being optional. We've discussed that making props unnecessarily optional makes it a bad developer experience when trying to use the ui-kit

- To add up to the previous point, when you have something like a type prop, a better approach is to use a union type strategy.  For example, right now the "single" type of file uploader can receive a "maxFiles" prop, which doesn't make conceptual sense. If you were to use a union type instead, we wouldn't need to pass irrelevant props in different logical cases. Same with the callbacks, I don't see why they should be optional

- I don't remember now why we approved the `text` prop is a strategy. Couldn't we just take those values from the dictionary? If it's because the text shown per category changes per lesson component, for example, then we can pass it from the outside, but then I don't see why would these fields be optional, whereas the `text` itself is not

- The download functionality is technically application logic, not ui-kit logic. It should be injected as a callback from the outside. The idea is that the app will decide what's the best way to handle downloads, as opposed to the UI kit enforcing this, and this will be more crucial in the course part of the app

- The upload callback should be a single one in my opinion, I don't see a reason to have two

alebg commented on packages/ui-kit/lib/components/drag&drop/file-uploader.tsx:
```diff
@@ -0,0 +1,218 @@
+  serverData?: UploadResponse;
+};
+
+export type EnhancedUploadFileProps = {
```
confusing naming. Why "Enhanced"?

jatinrise commented:
This commit includes the following changes:

- Updated ImageUploader and VideoUploader components to use a unified props structure for single and multiple uploads.
- Integrated localization support using getDictionary for text elements in the uploaders.
- Enhanced file upload handling with improved error management and state updates.
- Modified the ProfessionalInfo and ProfileInfo components to align with the new uploader structure.
- Updated storybook stories for `FileUploader`,` ImageUploader`, and `VideoUploader` to include locale options and better demonstrate functionality.
- Added tests for DragAndDrop component to validate file size restrictions and error handling.

jatinrise commented:
Hey @alebg, I'm not able to see the inline comment in my file, which is why I’ve added the screenshot below. So, I'll provide a point-wise explanation for each comment instead.
<img width="472" alt="Screenshot 2025-05-05 at 11 45 14 AM (1)" src="https://github.com/user-attachments/assets/75c0aa37-b934-441a-aed2-1207d6e28182" />

1. Yes, you're right that the type props shouldn't be optional. That's why I removed the optional properties and also used a union type for each prop.
2. have removed the` onDownload `function from the UI kit and added it as a callback.
3. Previously, I was passing two props: onUpload and onImageUpload callbacks. Now, I’ve combined them into a single callback called `onImageUpload`, which is no longer optional.
4. I have removed the `text` as a prop and localized the hardcoded text using the dictionary.

jatinrise commented:
Hey @alebg, the above commit includes the following changes:

1. I’ve created a single component to handle all three cases (file, image, and video) using a variant prop. You can specify the accepted file types using the acceptedFiles prop as an array.
2. To give a better idea of how the loading state will look, I’ve added a setTimeout to simulate a delay and show the UI.
3. I’ve also added error handling and its corresponding UI.

alebg commented:
Hi @jatinrise, thanks. Just a remark before I review your latest changes, this PR is very messy now. Maybe because you did a merge commit and solved conflicts?

![image](https://github.com/user-attachments/assets/8adcbb80-92d6-475d-af69-c2d42ec7c1d9)

With a lot of files being changed:

![image](https://github.com/user-attachments/assets/3d1f31d5-fb49-4b58-b89d-df0e566d5c2f)

But it's hard to tell just like that if all the changes in this PR to the non-relevant files to this task are keeping what was already done. Could you please create a new clean branch and put your work there?

In the future, as explained since the beginning, the preferred way to update your branches is through `git rebase main`. Please keep this in mind.

Also, unless you need something that was merged into main, you can just not do any merge commits or rebasing and let me handle this once I finish reviewing, that's how we've been operating the past couple of weeks.
